okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -33,7 +33,10 @@
 final class RealCall implements Call {
   final OkHttpClient client;
   final RetryAndFollowUpInterceptor retryAndFollowUpInterceptor;
-  final EventListener eventListener;
+
+  // There is a cycle between the Call and EventListener that makes this awkward. This will be set
+  // after we create the call instance then create the event listener instance.
+  volatile EventListener eventListener;
```

Comment:
At first I didn't have this, then I wondered what would happen if someone did fancy things in the EventListener.Factory.create method, i.e. something concurrent.

Also, it's unclear how to handle the case of someone calling `execute` from within the EventListener.Factory.create method. Any thoughts?